### PR TITLE
PixelsService: use Memoizer.failIfMissing

### PR DESCRIPTION
**Pull request by joshmoore**
To prevent DoS'ing a server, this flag can be set which
requires that all files have a valid cached Memoizer. A
next step will be to allow selectively permitting access
based on some flag.

See https://github.com/openmicroscopy/bioformats/pull/2435

NB: The boolean added to PixelsService should likely be refactored
    into a more general "options" class to reduce future method
    clutter.

### CODE REVIEW

File: components/romio/src/ome/io/nio/PixelsService.java [hunk 1]
```diff
@@ -740,11 +767,12 @@ protected void handleMissingPyramid(Pixels pixels,
      */
```
Comment (mtbc) [inline components/romio/src/ome/io/nio/PixelsService.java]: add to javadoc (this comment does not block the PR)

File: components/server/src/ome/security/SecuritySystem.java [hunk 2]
```diff
@@ -12,6 +12,8 @@
 // Third-party libraries
 
 // Application-internal dependencies
+import java.util.Map;
```
Comment (mtbc) [inline components/server/src/ome/security/SecuritySystem.java]: is not application-internal but again, non-blocking to this PR

**Comment by joshmoore** [comment]
Merging for IDR-0.0.5. As mentioned in the description, this may require further API work on the mainline.

**Comment by jburel** [comment]
--rebased-to #5106

**Pull request merged**